packages/cli/src/config/settings.ts

```diff
@@ -67,6 +67,36 @@ export interface Settings {
   // Add other settings here.
 }
 
+// The allSettings `Record<keyof Settings, undefined>`,
+// ensures that any changes to the `Settings` interface will result in a
+// compile-time error if this object is not also updated.
+const allSettings: Record<keyof Settings, undefined> = {
```

Comment:
Ideally we have a JSON schema of all supported keys and their expected values. Then we can just validate it using the newly available `SchemaValidator` class in the core package.

Reply:
agree a JSON schema for this would be ideal.